packages/react-router/index.tsx

```diff
@@ -704,6 +704,13 @@ export function createRoutesFromChildren(
       return;
     }
 
+    invariant(
+      element.type === Route,
+      `[${
+        typeof element.type === "string" ? element.type : element.type.name
```

Comment:
Here's why I'm checking the type: https://codesandbox.io/s/why-check-the-type-key-when-statically-analyzing-jsx-output-78rqp?file=/src/index.tsx